extensions/mssql/src/reactviews/pages/TableExplorer/TableDataGrid.tsx

```diff
@@ -251,6 +255,46 @@ export const TableDataGrid = forwardRef<TableDataGridRef, TableDataGridProps>(
             }
         }, [failedCells]);
 
+        // Sync deleted rows from props to ref and apply CSS classes
+        useEffect(() => {
+            if (deletedRows !== undefined) {
+                deletedRowsRef.current = new Set(deletedRows);
+
+                // Set up row metadata to apply CSS class to deleted rows
+                if (reactGridRef.current?.dataView) {
+                    const dataView = reactGridRef.current.dataView;
+
+                    // Store the original getItemMetadata if it exists
+                    const originalGetItemMetadata = dataView.getItemMetadata;
+
+                    // Override getItemMetadata to add CSS class for deleted rows
+                    dataView.getItemMetadata = function (row: number) {
+                        // Call original metadata function if it exists
+                        const item = dataView.getItem(row);
+                        let metadata = originalGetItemMetadata
+                            ? originalGetItemMetadata.call(this, row)
+                            : null;
+
+                        // Check if this row is deleted
+                        if (item && deletedRowsRef.current.has(item.id)) {
+                            metadata = metadata || {};
+                            metadata.cssClasses = metadata.cssClasses
+                                ? `${metadata.cssClasses} deleted-row`
+                                : "deleted-row";
+                        }
+
+                        return metadata;
+                    };
+
+                    // Force grid to re-render with new metadata
+                    if (reactGridRef.current?.slickGrid) {
+                        reactGridRef.current.slickGrid.invalidate();
+                        reactGridRef.current.slickGrid.render();
```

Comment:
hi there, `grid.invalidate()` is already calling `grid.render()` internally so there's no need to call it `render()` twice. I realized that my docs provided the wrong info, so I just opened a PR in my own project to reflect this: https://github.com/ghiscoding/slickgrid-universal/pull/2241